packages/amplify-e2e-tests/src/__tests__/geo-populate.test.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  createNewProjectDir,
+  deleteProject,
+  deleteProjectDir,
+  initJSProjectWithProfile,
+  addAuthWithDefault,
+  updateAuthAddUserGroups,
+  amplifyPushWithoutCodegen,
+  addGeofenceCollectionWithDefault,
+  populateGeofencesWithDefault,
+} from 'amplify-e2e-core';
+import { existsSync } from 'fs';
+import path from 'path';
+
+describe('amplify geo populate', () => {
+  let projRoot: string;
+  beforeEach(async () => {
+    projRoot = await createNewProjectDir('geo-populate-test');
+  });
+
+  afterEach(async () => {
+    const metaFilePath = path.join(projRoot, 'amplify', '#current-cloud-backend', 'amplify-meta.json');
+    if (existsSync(metaFilePath)) {
+      await deleteProject(projRoot);
+    }
+    deleteProjectDir(projRoot);
+  });
+
+  it('should throw error if there is no geofence collection provisioned', async () => {
+    await initJSProjectWithProfile(projRoot, {});
+    await addAuthWithDefault(projRoot);
+    await updateAuthAddUserGroups(projRoot, ['admin']);
+    await addGeofenceCollectionWithDefault(projRoot);
+    await expect(populateGeofencesWithDefault(projRoot)).rejects.toThrowError();
+  });
+  it('should upload a valid GeoJSON file with auto-generated root level ID to provisioned geofence collection', async () => {
+    await initJSProjectWithProfile(projRoot, {});
+    await addAuthWithDefault(projRoot);
+    await updateAuthAddUserGroups(projRoot, ['admin']);
+    await addGeofenceCollectionWithDefault(projRoot);
+    await amplifyPushWithoutCodegen(projRoot);
+    await expect(populateGeofencesWithDefault(projRoot, { geoJSONFileName: 'valid-root-level-id.json' })).resolves.not.toThrow();
```

Comment:
We might need to also verify that the collection now has the geofences in it by using `https://docs.aws.amazon.com/location-geofences/latest/APIReference/API_ListGeofences.html`. You can do this as an enhancement in a different PR if you would like to.